scene/gui/tab_bar.cpp

```diff
@@ -1367,14 +1367,14 @@ Variant TabBar::get_drag_data(const Point2 &p_point) {
 }
 
 bool TabBar::can_drop_data(const Point2 &p_point, const Variant &p_data) const {
-	if (!drag_to_rearrange_enabled) {
+	if (!drag_to_rearrange_enabled && !switch_on_hover) {
 		return Control::can_drop_data(p_point, p_data); // Allow stuff like TabContainer to override it.
 	}
 	return _handle_can_drop_data("tab_bar_tab", p_point, p_data);
 }
 
 void TabBar::drop_data(const Point2 &p_point, const Variant &p_data) {
-	if (!drag_to_rearrange_enabled) {
+	if (!drag_to_rearrange_enabled && !switch_on_hover) {
```

Comment:
```suggestion
	if (!drag_to_rearrange_enabled) {
```

This isn't needed in drop data.